### DISCUSSION TITLE
Understanding Folder Structures, Attachments

### DISCUSSION THREAD

**Discussion by DillBaggins** [redimp/otterwiki#108] [General]
First off, this looks like a really nice wiki manager, thank you to all involved in its development. I am playing around with otterwiki to create a wiki for my research group and in playing around I have had the following questions arise:

1. Understanding folder structures and the possibility of using nested folders in my wiki
2. Creating wiki entries directly in a file manager instead of through the GUI
3. Understanding where attachments uploaded through the GUI are stored

--- 
## 1) Understanding folder structures and the possibility of using nested folders in my wiki
I am looking to understand the workflow around having a folder structure to my wiki

My goal is to have related pages clustered in folders while still being accessible from anywhere else in the wiki and being indexed in the A-Z list. The structure I am looking for would look something like this:  
- home.md
- projects/
   - projectA/
     - attachments/
     - projectA_homepage.md
     - page1.md
     - page2.md
   - projectB/
- admin/

Playing around with it has yielded mixed results. Folders are showing up in the A-Z page index. My goal would be for only the pages themselves to show. 

Is this something that is achievable using otterwiki, either through the UI or by manually moving files around in the repository using a file manager?

--- 
## 2) Creating wiki entries directly in a file manager instead of through the GUI
I have been able to create wiki entries through my file manager and open them through the page index. However, the system breaks when attempting to make any edits (I am gathering this is due to the fact that they have not been git tracked). 

Is this a limitation of the system or is there a workaround? 

---
## 3)  Understanding where attachments uploaded through the GUI are stored

I have uploaded a couple of attachments to the preconfigured homepage but I am unable to find them in the file manager. Where in the repository are attachments saved by default?

**Comment by redimp** [comment]
Hey @DillBaggins, thank you for your intrest in An Otter Wiki.

## 1. The file- and directory structure

All data is stored as plain files in a git repository, pages are markdown files ending in `.md`), attachments to pages are stored in directories with the same name as the page (without the `.md`). Per default the filenames are all lowercase, this setting can be changed with the setting `RETAIN_PAGE_NAME_CASE` see [Configuration/Content and editing preferences](https://otterwiki.com/Configuration#content-and-editing-preferences). This might be useful, when editing files locally.

Your example would look like this
```
home.md
projects.md
projects/
projects/project a.md
projects/project a/page1.md
projects/project a/page2.md
projects/project a/example.png
projects/project b.md
admin.md
```
I've just created this example on <https://demo.otterwiki.com> the sidebar looks like this:

<img width="182" alt="image" src="https://github.com/redimp/otterwiki/assets/112056/bb6107fd-7c83-49c3-be9c-7cf5e965efe7">


## 2. Creating wiki entries directly in a file manager instead of through the GUI

You can create and modify the files in the local filesystem, just make sure to add your changes to the git repository.

Depending on your workflow, a better option might be to enable the `GIT_WEB_SERVER`, clone the Wiki, edit locally and push your changes.

## 3. Understanding where attachments uploaded through the GUI are stored

As mentioned above, the attachments are stored in a directory with the same name as the page, e.g. attaching `example.png` to `Home` will create the file in `home/example.png`.

**Reply by DillBaggins** [reply]
Hi @redimp 

Thank you very much for the thorough and detailed response. That answers all my questions perfectly.
